src/commcare_cloud/ansible/roles/bootstrap-users/tasks/main.yml

```
@@ -83,7 +83,7 @@
   assert:
     that:
       - '"{{ lookup("file", authorized_keys_dir ~ ssh_user ~ ".pub" ) }}" in ssh_pubkeys_string'
-  when: "'ansible' not in ssh_user"
+  when: ssh_user is not none
```

Does this actually work? I would expect that "none" must be capitalized (`None`) to be valid Python code. If it did not throw an error then maybe Ansible is doing something funny here (expression evaluates to false due to `NameError` or something like that), possibly causing the task to always be skipped?

---

@millerdev we are seeing this error when used `None` and it's not skipping every time, with `update-users` it is working as expected. 

when tags=bootstrap-users/users
```
TASK [bootstrap-users : Verifying ssh_user key in the ssh_pubkeys_string list] *********************************************************************************************************************************
task path: /tmp/commcare-cloud/src/commcare_cloud/ansible/roles/bootstrap-users/tasks/main.yml:82
fatal: [127.0.0.1]: FAILED! => {"msg": "The conditional check 'ssh_user is not None' failed. The error was: template error while templating string: no test named 'None'. String: {% if ssh_user is not None %} True {% else %} False {% endif %}\n\nThe error appears to be in '/tmp/commcare-cloud/src/commcare_cloud/ansible/roles/bootstrap-users/tasks/main.yml': line 82, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"Verifying ssh_user key in the ssh_pubkeys_string list\"\n  ^ here\n"}
```

---

Is there an error with `none` spelling? If not, what is the output of the task?

---

Ansible behavior here is confounding and, to add insult to injury, difficult to debug. After thrashing around for a while I think I found a solution that works
```suggestion
  when: (ssh_user)  # https://medium.com/opsops/wft-bool-filter-in-ansible-e7e2fd7a148f
```

---

thanks Daniel. I have updated the file.